src/app/gene-browser/gene.ts

```diff
@@ -170,6 +190,11 @@ export class Gene {
         ]);
       }
     }
+    this.collapsedTranscripts = collapseMultipleTranscripts(
+      new Array(...this.chromosomes).map(chrom => chrom[0]), this.transcripts
+    );
+    this.allSegments = collapseSegments(transcripts).segments;
```

Comment:
You can avoid having to call `collapseSegments` on `transcripts` twice if you store it in a temp var